src/metabase/driver/sql/query_processor.clj

```diff
@@ -1332,15 +1332,15 @@
            (not (uuid? (->honeysql driver arg)))
              ;; Check for inlined values
            (not (= (:database-type (h2x/type-info (->honeysql driver arg))) "uuid")))
-    [::cast field "varchar"]
+    [::cast field "varchar(36)"]
```

Comment:
I wonder what happens if a DB uses a different (strange) textual format to represent UUIDs. Is it really important to be specific here?